benchexec/tools/coccinelle.py

```diff
@@ -0,0 +1,64 @@
+# This file is part of BenchExec, a framework for reliable benchmarking:
+# https://github.com/sosy-lab/benchexec
+#
+# SPDX-FileCopyrightText: 2007-2020 Dirk Beyer <https://www.sosy-lab.org>
+#
+# SPDX-License-Identifier: Apache-2.0
+import logging
+import os
+import subprocess
+
+import benchexec.result as result
+from benchexec.tools.template import ToolNotFoundException, BaseTool2
+
+
+class Tool(BaseTool2):
+    """
+    Tool info for Coccinelle
+    (https://github.com/coccinelle/coccinelle).
+    """
+
+    REQUIRED_PATHS = ["standard.h" "ocaml" "standard.iso" "ocaml"]
+
+    def name(self):
+        return "Coccinelle"
+
+    def executable(self, tool_locator):
+        exe = tool_locator.find_executable("spatch")
+        dir_name = os.path.dirname(exe)
+        logging.debug("Looking in %s for spatch", dir_name)
+        for _, dir_names, file_names in os.walk(dir_name):
+            if "spatch" in file_names:
+                return exe
+            break
+        msg = (
+            f"ERROR: Did find a spatch in {os.path.dirname(exe)} "
+            f"but no 'spatch' directory besides it"
+        )
+        raise ToolNotFoundException(msg)
+
+    def version(self, executable):
+        return self._version_from_tool(executable)
+
+    def cmdline(self, executable, options, task, resource_limits):
+        return [executable] + options + list(task.input_files) + [task.property_file]
```

Comment:
Does this tool support more than one input file? If not, please use `task.single_input_file` in order to produce an appropriate error message.

What is used as a property file here?

Note that the current code will lead to a crash if no property file is given, because then `task.property_file` will be `None`.

Reply:
Thanks for the informative comments!
So the reason I am using `task.property_file` is because Coccinelle needs a .cocci template file in order to inject a task with a fault. And I am using the tool with Coveriteam and pass the template as a property file. 
I have modified the code now to consider a case of `None` in `task.property_file`.

Reply:
Thanks.

I am not sure whether the .cocci template really fits as a property file or whether it would be easier to understand if you just pass it as an input parameter of the tool (in `options`), but for the current state of the tool-info module it doesn't matter, so I am fine with it.